edxval/api.py

```diff
@@ -365,3 +373,99 @@ def copy_course_videos(source_course_id, destination_course_id):
             video=video,
             course_id=destination_course_id
         )
+
+
+def export_to_xml(edx_video_id):
+    """
+    Exports data about the given edx_video_id into the given xml object.
+
+    Args:
+        edx_video_id (str): The ID of the video to export
+
+    Returns:
+        An lxml video_asset element containing export data
+
+    Raises:
+        ValVideoNotFoundError: if the video does not exist
+    """
+    video = _get_video(edx_video_id)
+    video_el = Element(
+        'video_asset',
+        attrib={
+            'client_video_id': video.client_video_id,
+            'duration': unicode(video.duration),
+        }
+    )
+    for encoded_video in video.encoded_videos.all():
+         SubElement(
+            video_el,
+            'encoded_video',
+            {
+                name: unicode(getattr(encoded_video, name))
+                for name in ['profile', 'url', 'file_size', 'bitrate']
+            }
+        )
+    # Note: we are *not* exporting Subtitle data since it is not currently updated by VEDA or used
+    # by LMS/Studio.
+    return video_el
+
+
+def import_from_xml(xml, edx_video_id, course_id=None):
+    """
+    Imports data from a video_asset element about the given edx_video_id.
+
+    If the edx_video_id already exists, then no changes are made. If an unknown
+    profile is referenced by an encoded video, that encoding will be ignored.
+
+    Args:
+        xml: An lxml video_asset_data element containing import data
+        edx_video_id (str): The ID for the video content
+        course_id (str): The ID of a course to associate the video with
+            (optional)
+
+    Raises:
+        ValCannotCreateError: if there is an error importing the video
+    """
+    if xml.tag != 'video_asset':
+        raise ValCannotCreateError('Invalid XML')
+
+    try:
+        video = _get_video(edx_video_id)
+        logger.info(
+            "edx_video_id '%s' present in course '%s' not imported because it exists in VAL.",
+            edx_video_id,
+            course_id,
+        )
+        return
+    except ValVideoNotFoundError:
+        pass
+
+    data = {
+        'edx_video_id': edx_video_id,
+        'client_video_id': xml.get('client_video_id'),
+        'duration': xml.get('duration'),
+        'status': 'imported',
+        'encoded_videos': [],
+        'courses': [],
+    }
+    for encoded_video_el in xml.iterfind('encoded_video'):
+        profile_name = encoded_video_el.get('profile')
+        try:
+            profile = Profile.objects.get(profile_name=profile_name)
+        except Profile.DoesNotExist:
+            logger.info(
+                "Imported edx_video_id '%s' contains unknown profile '%s'.",
+                edx_video_id,
+                profile_name
+            )
+            continue
```

Comment:
IIRC, after discussion with the media team, we wanted to leave control over the profiles with them and not with course teams importing data. However, this may be an artifact from back when we didn't want to automatically create profiles with silly default values (for width, height, and extension). I will circle back with them for further discussion.